pubsub/subscriptions/main.go

```diff
@@ -167,6 +170,37 @@ func pullMsgsSettings(client *pubsub.Client, subName string) error {
 	return nil
 }
 
+func pullMsgsConcurrencyControl(client *pubsub.Client, subName string, numGoroutines int) error {
+	ctx := context.Background()
+	// [START pubsub_subscriber_concurrency_control]
+	sub := client.Subscription(subName)
+	sub.ReceiveSettings.NumGoroutines = numGoroutines
+	err := sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
+		fmt.Printf("Got message %q\n", string(msg.Data))
+		msg.Ack()
+	})
+	if err != nil {
+		return err
+	}
+	// [END pubsub_subscriber_concurrency_control]
+	return nil
+}
+
+func pullMsgsSynchronous(client *pubsub.Client, subName string) error {
+	ctx := context.Background()
+	// [START pubsub_subscriber_sync_pull]
+	sub := client.Subscription(subName)
+	sub.ReceiveSettings.Synchronous = true
+	err := sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
+		fmt.Printf("Got message %q\n", string(msg.Data))
+		msg.Ack()
+	})
+	if err != nil {
+		return err
+	}
+	// [END pubsub_subscriber_sync_pull]
+	return nil
+}
```

Comment:
We need a more elaborate sample here that does synchronous pull with lease management. The sample needs to show how to modify message ack deadline for long-running tasks. Here are the existing [Python](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/7d01d749fb616eb5332512413c47634b199b28e6/pubsub/cloud-client/subscriber.py#L293) and [Node](https://github.com/googleapis/nodejs-pubsub/blob/b9408118dbc5e4d8c580e23a586c0e19caeae0db/samples/subscriptions.js#L273) samples.

Reply:
Done. I noticed that the samples used in the docs are actually inconsistent. We want 2 samples in this case: one to show how to do simple Synchronous Pull, and another sample to show lease management. The Python/Node samples are missing the simple case, since [the docs](https://cloud.google.com/pubsub/docs/pull#synchronous_pull) actually say:

> Here is some sample code to pull and acknowledge a fixed number of messages:

This is hard to read for users who want to know _how_ to do sync pull. We should add another section in the docs referencing doing synchronous pull with lease management as you said.

Reply:
Let's come back to this after discussing with Ken/Kir.